permuta/perm.py

```diff
@@ -53,29 +62,31 @@ def __new__(cls, iterable=(), check=False):
         """
         return tuple.__new__(cls, iterable)
 
-    def __init__(self, iterable=(), check=False):
+    def __init__(self, iterable: Iterable = (), check: bool = False) -> None:
         # Cache for data used when finding occurrences of self in a perm
-        self._cached_pattern_details = None
+        self._cached_pattern_details: Optional[
+            List[Tuple[Optional[int], Optional[int], int, int]]
+        ] = None
         if check:
             self._init_checked()
 
-    def _init_checked(self):
+    def _init_checked(self) -> None:
         """Checks if a suitable iterable given when initialised."""
         used = [False] * len(self)
-        for value in self:
-            if not isinstance(value, numbers.Integral):
-                message = "'{}' object is not an integer".format(repr(value))
+        for val in self:
+            if not isinstance(val, numbers.Integral):
+                message = "'{}' object is not an integer".format(repr(val))
                 raise TypeError(message)
-            if not 0 <= value < len(self):
-                raise ValueError("Element out of range: {}".format(value))
-            if used[value]:
-                raise ValueError("Duplicate element: {}".format(value))
-            used[value] = True
+            if not 0 <= val < len(self):
+                raise ValueError("Element out of range: {}".format(val))
+            if used[val]:
+                raise ValueError("Duplicate element: {}".format(val))
+            used[val] = True
 
-    _to_standard_cache = {}
+    _to_standard_cache: ClassVar[Dict[Tuple, "Perm"]] = {}
 
     @classmethod
-    def to_standard(cls, iterable):
+    def to_standard(cls, iterable: Iterable) -> "Perm":
```

Comment:
`Iterable[int]` ?

Reply:
I think this is used from all sort of iterables. The example in the docstring even has one with a string.

Reply:
Seems reasonable even if in practice we only ever use int